File: 94-binary-tree-inorder-traversal/94. Binary Tree Inorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void inorder(TreeNode * root , vector<int>& sample) {
        if (root == nullptr) {
            return; 
        }
        inorder(root -> left, sample);
        sample.push_back(root -> val);
        inorder(root -> right, sample); 
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> sample; 
        inorder(root, sample);
        return sample; 
    }
};

File: 94-binary-tree-inorder-traversal/94. Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def collector(self, root, getter):
        if(root == None):
            return
        
        self.collector(root.left, getter)
        getter.append(root.val)
        self.collector(root.right, getter)
    
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.collector(root, result)
        return result
        